packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -999,56 +999,44 @@ function useMutableSource<Source, Snapshot>(
     subscribe,
   }: MutableSourceMemoizedState<Source, Snapshot>);
 
-  // Sync the values needed by our subscribe function after each commit.
+  // Sync the values needed by our subscription handler after each commit.
   dispatcher.useEffect(() => {
-    const didGetSnapshotChange = !is(refs.getSnapshot, getSnapshot);
     refs.getSnapshot = getSnapshot;
 
     // Normally the dispatch function for a state hook never changes,
-    // but in the case of this hook, it will change if getSnapshot changes.
-    // In that case, the subscription below will have cloesd over the previous function,
-    // so we use a ref to ensure that handleChange() always has the latest version.
+    // but this hook recreates the queue in certain cases  to avoid updates from stale sources.
+    // handleChange() below needs to reference the dispatch function without re-subscribing,
```

Comment:
I think if you go with this approach then you don't have to track `getSnapshot` or `setSnapshot` at all. Can use the closed over values, since `getSnapshot` is now a dep, and `setSnapshot` only changes when one of the other deps does.

Reply:
Edit: Got confused, thought you had removed the `getSnapshot` optimization. Never mind.

Reply:
No worries. This is something we can further optimize when we replace the two composed effects with our own pushed effect. Then we can just use one single effect and selectively unsubscribe/subscribe only when necessary.

Sounds like that will be a fun little cleanup refactor anyway. :smile: